tree/ntuple/v7/src/RNTupleProcessor.cxx

```diff
@@ -352,13 +349,28 @@ ROOT::Experimental::RNTupleJoinProcessor::RNTupleJoinProcessor(const RNTupleOpen
 
    fNEntries = fPageSource->GetNEntries();
 
-   if (models.empty())
-      fModel = fPageSource->GetSharedDescriptorGuard()->CreateModel();
-   else
-      fModel = models[0]->Clone();
+   for (const auto &auxNTuple : auxNTuples) {
+      fAuxiliaryPageSources.emplace_back(Internal::RPageSource::Create(auxNTuple.fNTupleName, auxNTuple.fStorage));
+      if (!joinFields.empty())
+         fJoinTables.emplace_back(Internal::RNTupleJoinTable::Create(joinFields));
+   }
+
+   if (!primaryModel)
+      primaryModel = fPageSource->GetSharedDescriptorGuard()->CreateModel();
+   if (auxModels.empty()) {
+      auxModels.resize(fAuxiliaryPageSources.size());
+   }
+   for (unsigned i = 0; i < fAuxiliaryPageSources.size(); ++i) {
+      if (!auxModels[i]) {
+         fAuxiliaryPageSources[i]->Attach();
+         auxModels[i] = fAuxiliaryPageSources[i]->GetSharedDescriptorGuard()->CreateModel();
+      }
+   }
+
+   SetModel(std::move(primaryModel), std::move(auxModels));
 
    fModel->Freeze();
-   fEntry = fModel->CreateEntry();
+   fEntry = fModel->CreateBareEntry();
```

Comment:
Can you test this with a bare model? I think in that case this change creates a bare entry without `BindValue` below.

A possible improvement could be to just use the default entry in case of a non-bare model (without carrying over the value pointers below), and only call `CreateEntry()` for a bare model.

Reply:
Done, see latest commit. I have actually been considering to use the default entry instead to simplify things, but since this should apply to the other processors then as well I will address this in a separate PR.

Reply:
Done (and indeed have to use a regular entry here). I was thinking the same for the default entry actually, that would simplify a lot of things. This would also require changes to the other processors to keep it consistent, so I will open a separate PR for this.